src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -336,7 +336,11 @@ - (id)copyWithZone:(NSZone *)zone
 
 - (BOOL)isEqualToClusterData:(MTRDeviceClusterData *)otherClusterData
 {
-    return [_dataVersion isEqual:otherClusterData.dataVersion] && [_attributes isEqual:otherClusterData.attributes];
+    if (!otherClusterData)
+        return NO;
+
+    return (otherClusterData.dataVersion && [_dataVersion isEqual:otherClusterData.dataVersion])
+        && (otherClusterData.attributes && [_attributes isEqual:otherClusterData.attributes]);
```

Comment:
This is again wrong: dataVersion is allowed to be nil, and if it's nil we should be doing the right checks.

On the other hand, attributes is not allowed to be nil, and we should not be worrying about it being nil.  We should fix `initWithCoder:` to guarantee it's not nil, just like `initWithDataVersion:attributes:` does.